branchtesting/src/main/java/branchexample/App.java

```diff
@@ -1,16 +1,23 @@
 package branchexample;
 
+import java.math.BigDecimal;
+
 public class App 
 {
+	public static String vulnerableString = new String();
+	
     public void aMethod( String[] args )
     {
-    	// TODO fix this
+    	double d = 1.1;
+
+    	BigDecimal bd1 = new BigDecimal(d); // Buggy
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use "BigDecimal.valueOf" instead. ([squid:S2111](https://sonarcloud.io/organizations/brian-cipollone-sonarsource-gith/rules#rule_key=squid%3AS2111))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.example.bcipollone%3Abranchtesting&pullRequest=12&issues=AWWImbQmXKENQn1SC_7M&open=AWWImbQmXKENQn1SC_7M)

[ISSUE_KEY]: AWWImbQmXKENQn1SC_7M